### PR TITLE
AB2-158:add option to enable/disable quick search

### PR DESCRIPTION
**Pull request by moaz-cliqz**

### CODE REVIEW

File: mozilla-release/mobile/android/app/src/main/res/xml/preferences_general.xml [hunk 1]
```diff
@@ -46,19 +46,24 @@
...
+        -->
         <CheckBoxPreference
             android:key="pref.search.block.adult.content"
             android:title="@string/pref_search_block_adult_content" />
         <CheckBoxPreference
             android:key="pref.search.auto.completion"
             android:title="@string/pref_search_auto_completion" />
-        <!-- Cliqz start -->
+        <CheckBoxPreference
+            android:key="pref.search.enable"
```
Comment (spacifici) [inline mozilla-release/mobile/android/app/src/main/res/xml/preferences_general.xml]: Please name everything related to `quicksearch` as `pref.quicksearch.xxx`

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java [hunk 2]
```diff
@@ -875,15 +875,14 @@ public boolean onInterceptMotionEvent(View view, MotionEvent event) {
...
-        // mBrowserSearchContainer = findViewById(R.id.search_container);
-        // mBrowserSearch = (BrowserSearch) getSupportFragmentManager().findFragmentByTag(BROWSER_SEARCH_TAG);
-        // if (mBrowserSearch == null) {
-        //     mBrowserSearch = BrowserSearch.newInstance();
-        //     mBrowserSearch.setUserVisibleHint(false);
-        // }
-        /* Cliqz end */
+         /* Cliqz start */
+        mPreferenceManager = new PreferenceManager(getBaseContext());
+         mBrowserSearchContainer = findViewById(R.id.search_container);
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java]: Indentation

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java [hunk 3]
```diff
@@ -260,12 +260,8 @@
...
     private static final String FIRSTRUN_UUID = "firstrun_uuid";
 
-    /* Cliqz start */
-    // @Cliqzers: we are not using the Mozilla search fragment. We may have to restore it later when
-    // we will implement our native search.
-    // private BrowserSearch mBrowserSearch;
-    // private View mBrowserSearchContainer;
-    /* Cliqz end */
+     private BrowserSearch mBrowserSearch;
+     private View mBrowserSearchContainer;
```
Comment (ravjit-cliqz) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java]: Something wrong with indentation here

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java [hunk 4]
```diff
@@ -875,15 +875,14 @@ public boolean onInterceptMotionEvent(View view, MotionEvent event) {
...
-        /* Cliqz start */
-        // mBrowserSearchContainer = findViewById(R.id.search_container);
-        // mBrowserSearch = (BrowserSearch) getSupportFragmentManager().findFragmentByTag(BROWSER_SEARCH_TAG);
-        // if (mBrowserSearch == null) {
-        //     mBrowserSearch = BrowserSearch.newInstance();
-        //     mBrowserSearch.setUserVisibleHint(false);
-        // }
-        /* Cliqz end */
+         /* Cliqz start */
+        mPreferenceManager = new PreferenceManager(getBaseContext());
```
Comment (ravjit-cliqz) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java]: Here also

File: mozilla-release/mobile/android/base/locales/en-US/cliqz_strings.dtd [hunk 5]
```diff
@@ -142,6 +142,8 @@
 <!ENTITY pref_search_complementary_engine_ecosia "Ecosia">
 <!ENTITY pref_search_complementary_engine_start_page "StartPage">
 <!ENTITY pref_search_complementary_engine_qwant "Qwant">
+<!ENTITY pref_search_enable "Enable Quick Search">
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/locales/en-US/cliqz_strings.dtd]: Naming here too

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java [hunk 6]
```diff
@@ -3432,13 +3434,16 @@ private void showBrowserSearch() {
...
     }
 
     private void hideBrowserSearch() {
         /* Cliqz start */
-        // if (!mBrowserSearch.getUserVisibleHint()) {
-        //     return;
-        // }
+        if (!mPreferenceManager.isQuickSearchEnabled()) {
+            if (!mBrowserSearch.getUserVisibleHint()) {
+                return;
```
Comment (ravjit-cliqz) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java]: You should merge both the "if" into one statement

Comment (moaz-cliqz) [reply mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java]: done

**Pull request merged**